### ISSUE TITLE
[BUG] JS template literal close indent issue

### ISSUE THREAD

**Issue by jpradelle** [yeonjuan/html-eslint#475]
Hi, I'm opening issue from this comment: https://github.com/yeonjuan/html-eslint/issues/462#issuecomment-3675348342

In rule @html-eslint/indent I found issues with template literal closing indent check.

Those are good
```javascript
  good() {
    return html`
      ${when(true, () => html`
        <b>ok</b>
      `)}

      ${when(
          true,
          () => html`
            <b>ok</b>
          `
      )}
    `;
  }
```

But for this case an error is reported:
```javascript
  bad() {
    return html`
      ${when(
          true,
          () => html`
            <b>ok</b>
          `)}
      ${when(
          true,
          () => html`
            <b>ok</b>`)}
    `;
  }
```
If I fix it, it does:
```javascript
  bad() {
    return html`
      ${when(
          true,
          () => html`
            <b>ok</b>
      `)}
      ${when(
          true,
          () => html`
      <b>ok</b>`)} <-- another error is reported after on this one asking for previous indent
    `;
  }
```

But I think we want to keep the 3 closing chars
```
`)}
```
at expected indent level of closing back quote, not closing bracket
The same way we can do for functions like this:
```javascript
  foo() {
    this.foo(
        this.bar,
        this.baz(
            this.baaz()));
    this.foo(
        this.bar,
        this.baz(
            this.baaz()
        ));
  }
```

**Comment by yeonjuan** [comment]
Hi @jpradelle Thank you for your report. Let me check

**Comment by yeonjuan** [comment]
Hi @jpradelle I've deployed the code with the bug fixed in the version below. Could you please verify if it's working as intended? If you're unable to verify it, please let me know. I'll conduct further testing before implementing it.
[0.53.0-alpha.0](https://www.npmjs.com/package/@html-eslint/eslint-plugin/v/0.53.0-alpha.0)
```
npm install -D @html-eslint/eslint-plugin@0.53.0-alpha.0
```

**Comment by jpradelle** [comment]
Hi,
You fixed properly the conflict on that case properly, no more error is reported here.
```
      ${when(
          true,
          () => html`
            <b>ok</b>`)}
```

But I'm still disturbed with the expected level of closing back quote.
No error is reported here, and this is what I expect:
```
      ${when(
          true,
          () => html`
            <b>ok</b>
          `
      )}
```
But in that case an error is reported here (Expected indentation of 6 space but found 10 space)
```
      ${when(
          true,
          () => html`
            <b>ok</b>
          `)}
```
Asking to replace to
```
      ${when(
          true,
          () => html`
            <b>ok</b>
      `)}
```

I still believe this should be valid (this is what is built by my IDE by default too):
```
      ${when(
          true,
          () => html`
            <b>ok</b>
          `)}
```

While testing, I also found another issue, closing back quote is not always checked:
```
      ${when(
          true,
          () => html`
            <p>ok</p>
    ` // <-- whatever indent, no error is reported
      )}
```

**Comment by yeonjuan** [comment]
Hi @jpradelle 
Thank you for your confirmation. The lack of checking indentation for backticks is intentional because they are not part of the HTML code.

**Comment by jpradelle** [comment]
But checking indent of backticks was the purpose of issue https://github.com/yeonjuan/html-eslint/issues/462 , and you made a fix for it, no ?

**Comment by yeonjuan** [comment]
@jpradelle 
What I modified in that issue was not the backtick itself, but the opening and closing parts of the template literal.(`${`, `}`). In html-eslint, this part is treated as if a template engine syntax (such as Handlebars) were being used.

The indentation of the backtick should probably align with the line where the tag of the tagged template is located. However, the intent of the tag part of a tagged template is not handled by html-eslint.

```js
   // We do not handle the intent of the "tag" part.
   tag`
  <div>...

`
```

**Comment by jpradelle** [comment]
Ok I get it.
But here there will a conflict with other rules that might check indent of closing backtick from another linter.

If I do that, using another linter checking backtick indent, this will be allowed by the other linter, but forbidden by html-eslint linter
```
      ${when(
          true,
          () => html`
            <b>ok</b>
          `)}
```
And doing that wil allowed by html-eslint linter, but forbidden by the other linter
```
      ${when(
          true,
          () => html`
            <b>ok</b>
      `)}
```

I think, as you don't check other content rules, you should check indent of closing bracket, only if this is the only element of the line.

Meaning this would be valid (html-eslint doesn't check it, this is other linter responsibility to check indent based on backtick indent.
```
      ${when(
          true,
          () => html`
            <b>ok</b>
          `)}
```
And this would be valid too, html-eslint requiring closing brack to be exactly at this level.
```
      ${when(
          true,
          () => html`
            <b>ok</b>
          `)
      }
```
